sanitize all electron.shell.openExternal calls

dimitropoulos opened the pull request:
partially closes INS-590

DMarby commented on packages/insomnia-app/app/common/misc.js:
```diff
@@ -467,3 +467,9 @@ export function snapNumberToLimits(value: number, min?: number, max?: number): n
 }
+
+export const sanitizedOpenExternal = (href: string) => {
+  if (href.startsWith('http')) {
```
Not sure if it might be safer to parse the URL and check the protocol here (`new URL(url).protocol`)?
But I think this is probably good as-is 👍

develohpanda replied:
Or maybe a regex check as well because it wasn't immediately apparent to me that https is supported: `/^https?/`

dimitropoulos replied:
love it! I went with https://github.com/Kong/insomnia/pull/3336/commits/23fe70fde2e16c9ecfce293bc85e058aa699fa22 since it seemed to be the most simple approach and doesn't require knowledge of regexes (in the aim to be more beginner friendly where possible, although I love me a good regex).

DMarby reviewed (approved):
Awesome! 🎉

dimitropoulos committed:
sanitize all electron.shell.openExternal calls

dimitropoulos committed:
adds lint rule to disallow future usage of `openExternal`

dimitropoulos committed:
updates to use URL constructor per review feedback

dimitropoulos committed:
moves restartApp with the rest of the electron-helpers

I noticed that this one was astray, but now that there's a home for this kind of helper, it should go with the others.
